BUILD.bazel

```diff
@@ -1,12 +1,44 @@
+load(
+    "//bazel:windows_dll_library.bzl",
+    "windows_dll_library",
+)
+
+# choose which library to use depending on the OS
+alias(
+  visibility = ["//visibility:public"],
+  name = "opentracing",
+  actual = select({
+    "@bazel_tools//src/conditions:windows": ":opentracing_windows",
+    "//conditions:default": ":opentracing_unix",
+  }),
+)
+
+windows_dll_library(
+  name="opentracing_windows",
+  srcs = glob(["src/**/*.cpp"], exclude = ["src/dynamic_load_unsupported.cpp", "src/dynamic_load_unix.cpp"]),
+  hdrs = glob(["include/opentracing/**/*.h"]) + [
+      ":include/opentracing/config.h",
+      ":include/opentracing/version.h",
+  ],
+  # so that source files can write opentracing/... instead of include/opentracing/...
+  includes=["include"],
+  deps = [
+    "//3rd_party:expected",
+    "//3rd_party:variant",
+  ],
+  # sets the OPENTRACING_EXPORTS precompiler variable while building DLL
+  # so that header "import symbols" tags become "export symbols" tags
+  copts=["/DOPENTRACING_EXPORTS"],
+)
+
```

Comment:
Would a windows user be able to also link statically to opentracing-cpp if they wanted to?

Reply:
For now, I'm going to hold off on supporting static linking because I think it will be a challenge. I was having a bunch of issues with this, and then found [this thread](https://github.com/bazelbuild/bazel/issues/1920) which explains how cc_binary isn't setup to build static libraries.

Reply:
No, we need opentracing to be a static library. There are existing builds that depend on that and will break if it's changed.

A thin archive is the default output for `cc_library`. That thread you link to is something different, a thick archive with all of the dependencies linked in -- that's not necessary. 

Just keep cc_library the same but select the correct dynamic_load source for windows.